### PR TITLE
Deprecate the to/from_recarray methods

### LINKED ISSUE #123: deprecate the `to_recarray` function
This function facilitated an old way that pyuvsim handled catalogs but it's not needed anymore and it's quite clunky.

Note that it is currently used in the `write_text_catalog` function but we can rename it to something internal for that method.

### PR DESCRIPTION
**Pull request by bhazelton**
## Motivation and Context

closes #123 

### Other Change Checklist

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyradiosky/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.
- [x] Any new or updated docstrings use the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] I have added tests to cover any changes.
- [x] My change includes a breaking change
  - [x] My change includes backwards compatibility and deprecation warnings (if possible).
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyradiosky/blob/main/CHANGELOG.md) if appropriate.

### CODE REVIEW

**Comment by bhazelton** [comment]
The pyuvsim tests will be fixed when https://github.com/RadioAstronomySoftwareGroup/pyuvsim/pull/393 is merged.

File: pyradiosky/skymodel.py [hunk 1]
```diff
@@ -5411,7 +5438,9 @@ def read_votable_catalog(
     )
 
     if return_table:
-        return skyobj.to_recarray()
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
+            return skyobj.to_recarray()
```
Comment (mkolopanis) [inline pyradiosky/skymodel.py:5443]: I see that this function as a whole is deprecated, So we're not too worried about issuing yet another deprecation inside of it I presume?

Comment (bhazelton) [reply pyradiosky/skymodel.py:5443]: exactly

Comment (bhazelton) [reply pyradiosky/skymodel.py:5443]: Oh, maybe I misunderstood your question. I just figured we didn't need multiple deprecation warnings, but I can remove this filter.

Comment (mkolopanis) [reply pyradiosky/skymodel.py:5443]: Oh no I think you understood it correctly. There's a few like this so I just wanted to double check

File: pyradiosky/tests/test_skymodel.py [hunk 2]
```diff
@@ -2098,13 +2098,23 @@ def test_order_healpix_to_sky(healpix_data, hpx_order):
...
 
-@pytest.mark.filterwarnings("ignore:recarray flux columns will no longer be labeled")
 def test_healpix_recarray_loop(healpix_disk_new):
 
     skyobj = healpix_disk_new
-    skyarr = skyobj.to_recarray()
+    with uvtest.check_warnings(
+        DeprecationWarning,
+        match=[
+            "The to_recarray method is deprecated and will be removed in 0.3.0.",
```
Comment (aelanman) [inline pyradiosky/tests/test_skymodel.py:2107]: To confirm -- Is the plan to deprecate `array_to_skymodel` in 0.2.0 and then `to_recarray` in 0.3.0?

**Review by aelanman — APPROVED** [review]
Looks good to me! Just one minor question

File: pyradiosky/tests/test_skymodel.py (hunk 2 above)
Comment (bhazelton) [reply pyradiosky/tests/test_skymodel.py:2107]: yes. There's also a `from_recarray` which was the replacement for `array_to_skymodel`, which will also be deprecated in 0.3.0.

**Pull request merged**